src/command.rs

```diff
@@ -189,7 +189,12 @@ impl EccCommand {
                 put_cmd!(bytes, ATCA_INFO, 0, 0);
             }
             Self::GenKey { key_type, slot } => {
-                put_cmd!(bytes, ATCA_GENKEY, u8::from(key_type), (*slot as u16) << 8);
```

Comment:
This is very odd? The `put_cmd` macro uses [put_u16](https://docs.rs/bytes/latest/bytes/buf/trait.BufMut.html#method.put_u16)  which is part of the BufMut trait and states that it writes the given u16 in big endian order. 

So in this specific fix that would mean the << 8 above turns it into big endian and then to_be turns it back into little endian and, finally put_u16 turns it back into big endian form.